src/consensus/pbft_manager.cpp

```diff
@@ -1271,9 +1271,51 @@ std::pair<blk_hash_t, bool> PbftManager::proposeMyPbftBlock_() {
 
   uint64_t propose_pbft_period = pbft_chain_->getPbftChainSize() + 1;
   addr_t beneficiary = node_addr_;
+
+  // get DAG block and transaction order
+  auto dag_block_order = dag_mgr_->getDagBlockOrder(dag_block_hash);
+  if (dag_block_order.second->empty()) {
+    LOG(log_er_) << "DAG anchor block hash " << dag_block_hash << " getDagBlockOrder failed in propose";
+    assert(false);
+    return std::make_pair(NULL_BLOCK_HASH, true);
```

Comment:
what is the logic behind return true on failures ? On line 1220 we return false for example...

Reply:
I have removed the return after assert but I also do not understand the meaning of the flag in the return values of this method. Lets talk about this